src/main/scala/org/apache/daffodil/DaffodilPlugin.scala

```diff
@@ -290,9 +295,12 @@ object DaffodilPlugin extends AutoPlugin {
      */
     libraryDependencies ++= {
       if (daffodilBuildsLayer.value) {
-        Seq(
-          "org.apache.daffodil" %% "daffodil-runtime1-layers" % daffodilVersion.value % "provided"
+        val charsetDep = Map(
+          ">=4.0.0 " -> "org.apache.daffodil" % "daffodil-core_3" % daffodilVersion.value % "provided",
+          "=3.11.0 " -> "org.apache.daffodil" % "daffodil-runtime1-layers_2.13" % daffodilVersion.value % "provided",
+          "<=3.10.0 " -> "org.apache.daffodil" % "daffodil-runtime1-layers_2.12" % daffodilVersion.value % "provided"
         )
+        filterVersions(daffodilVersion.value, charsetDep)
       } else {
         Seq()
       }
```

Comment:
Note that if DAFFODIL-2994 is implemented to merge daffodil-udf into daffodil-core, then the daffodilBuildsUDF case below will need a similar change. Nothing to do for now, but something to keep in mind.